proposals/native-lock.md

```diff
@@ -0,0 +1,83 @@
+# Native lock
+
+## Summary
+[summary]: #summary
+
+Special-case how `System.Threading.Lock` interacts with the `lock` keyword (calling its `EnterLockScope` method under the hood).
+Add static analysis warnings to prevent accidental misuse of the type where possible.
+
+## Motivation
+[motivation]: #motivation
+
+.NET 9 is introducing a new [`System.Threading.Lock` type](https://github.com/dotnet/runtime/issues/34812)
+as a better alternative to existing monitor-based locking.
+The presence of the `lock` keyword in C# might lead developers to think they can use it with this new type.
+Doing so wouldn't lock according to the semantics of this type but would instead treat it as any other object and would use monitor-based locking.
+
+## Detailed design
+[design]: #detailed-design
+
+Semantics of the lock statement ([§13.13](https://github.com/dotnet/csharpstandard/blob/9af5bdaa7af535f34fbb7923e5406e01db8489f7/standard/statements.md#1313-the-lock-statement))
+are changed to special-case the `System.Threading.Lock` type:
+
+> A `lock` statement of the form `lock (x) { ... }`
+>
+> 1. **where `x` is an expression of type `System.Threading.Lock` is precisely equivalent to:**
+>    ```cs
+>    using (x.EnterLockScope())
```

Comment:
We should specify if extension methods are or are not allowed. The wording of the spec as is implies that extension methods are allowed but given we're special casing a type in the BCL I imagine we don't want to allow that as it would be additional cost for no benefit.

Reply:
Yes, I think we want to require the `Lock` type to have a specific shape and we want to use just the well-known members from that shape. Not sure if the speclet should say what happens if some `Lock` member is missing from the expected shape (fall back to the normal lock; issue an error; leave that implementation-specific).